### PR TITLE
remove asyncComputed from MergeTable

### PR DESCRIPTION
**Pull request by RayBB**
Contributes to #7547 by helping us be able to upgrade to Vue 3!

Removes asyncComputed, which doesn't support Vue3. The beta library wasn't working either. 

We don't need asyncComputed because the async values don't need to be "computed" more than once. We only have to wait for `records` before async fetching bookshelves, lists, editions, and ratings. Each of those 4 only need a tiny bit if post processing but never change after the intital request.

The new approach is to load those four right after we get the `records` in the created lifecycle step.

I'll walk through the code in the video.

### Technical

This PR is a big refactor but none of the logic was not changed, just moved.

### Testing

This really only impacts the loading of data so if you can see everything looks the same then we should be good to go.

### Stakeholders

@jimchamp

### CODE REVIEW

File: openlibrary/components/MergeUI.vue [hunk 1]
```diff
@@ -11,10 +11,11 @@
             <button class="reject-btn" v-if="showRejectButton" @click="rejectMerge">Reject Merge</button>
         </div>
         <div id="diffs-toggle">
-        <label>
-            <input type="checkbox" title="Show textual differences" v-model="show_diffs" />
-            Show text diffs
-        </label>
+            <label>
```
Comment (RayBB) [inline openlibrary/components/MergeUI.vue:14]: This is addressed in #8385 but was needed for local development.

**Comment by mekarpeles** [comment]
Howdy @RayBB, if @cdrini can help unblock you this week, that would be amazing -- if by Friday we don't have feedback, I'd like @jimchamp to help take this over the finish line, even if the solution may not be ideal (we can always fix things moving forward and iterate but blockers are going to prevent us from forward progress towards vue3 and we're getting down to the wire: December 31st)

**Comment by RayBB** [comment]
I talked with @cdrini about this PR very briefly on the phone. I addressed his comments best I could to improve naming.

I don't have a link now but this approach was mentioned in a forum about how to deal with a situation like this. Basically saying if you're not fetching more than once then fetch it in the `created` event hook to keep things simple.

I don't love this approach as asyncomputed is a bit less code but I think it's better than migrating to another library for this simple use case. If someone really wants to try to get the beta version of the asyncomputed library working then go for it but I didn't have luck. Just vague error messages.

As far as this PR goes: **Modify it as you see fit (or make a new branch)** or close it if you wanna migrate to the other library but I don't have energy this month to refactor it more. I updated the package.json again so it can be merged.

**Comment by RayBB** [comment]
@jimchamp @mekarpeles let me know if there are any further questions on this.

**Comment by cdrini** [comment]
Howdy! I'm taking a look at updating the source package to Vue 3 ; it seems like a not complicated updated, and it would help a toooon of people who also rely on this package :) https://github.com/foxbenjaminfox/vue-async-computed/pull/124

I'd like to see if the maintainer will give a response on this one. Help appreciated on the last question/testing on that PR!

**Comment by cdrini** [comment]
I'll drop this to p2 and revisit after this week! Hopefully we'll get a response from the maintainer

**Comment by cdrini** [comment]
Closing this as we were able to get in contact with vue-async-computed and migrated them to Vue 3! We'll just have to update to the latest vue-async-computed when we migrate to Vue 3 :+1:

**Comment by RayBB** [comment]
Updated the vue 3 issue to reflect this.

**Pull request closed without merging**